### DISCUSSION TITLE
Why is there a setnonblocking function that is unused?

### DISCUSSION THREAD

**Discussion by JohnZ622** [esnet/iperf#1789] [Q&A]
I see setnonblocking function in net.c but it is invoked anywhere.  A unfinished feature or dead code?  It's not necessary to use non-blocking sockets here right?

**Comment by bmah888** [comment]
More like dead code. IIRC, we needed to use non-blocking sockets when iperf3 had a single-threaded design, in order to avoid hanging the main (and only) thread. That's not really a thing for the multi-threaded design used in iperf-3.16 and later.
